internal/testdata/source/javascript/statements.js

```diff
@@ -105,21 +105,117 @@ function LabeledWhileStatement() {
 }
 
 function SwitchStatement() {
+    console.log("switch entry")
     let fruits = 'Oranges'
 
     switch (fruits) {
         case 'Oranges':
-            console.log('Oranges')
+            console.log("switch case 1")
             break
         case 'Mangoes':
-            console.log('Mangoes')
+            console.log("switch case 2")
             break
         case 'Papayas':
-            console.log('Papayas')
+            console.log("switch case 3")
             break
         default:
-            console.log('No fruits')
+            console.log("switch case default")
     }
+
+    console.log("switch done")
+}
+
+function SwitchStatementWithoutDefault() {
+    console.log("switch entry")
+    let fruits = 'Oranges'
+
+    switch (fruits) {
+        case 'Oranges':
+            console.log("switch case 1")
+            break
+        case 'Mangoes':
+            console.log("switch case 2")
+            break
+        case 'Papayas':
+            console.log("switch case 3")
+            break
+    }
+
+    console.log("switch done")
+}
+
+function SwitchStatementOnlyDefault() {
+    console.log("switch entry")
+    let fruits = 'Oranges'
+
+    switch (fruits) {
+        default:
+            console.log("switch case default")
+    }
+
+    console.log("switch done")
+}
+
+
+function SwitchStatementOnlyOneCase() {
+    console.log("switch entry")
+    let fruits = 'Oranges'
+
+    switch (fruits) {
+        case 'Oranges':
+            console.log("switch case 1")
+            break
+    }
+
+    console.log("switch done")
+}
+
+function SwitchStatementWithBadNodesAndDefault() {
+    console.log("switch entry")
+    let fruits = 'Oranges'
+
+    switch (fruits) {
+        // case 'Oranges':
+        //     console.log("switch case 1")
+        //     break
+        default:
+            console.log("switch case default")
+    }
+
+    console.log("switch done")
+}
+
+function SwitchStatementWithBadNode() {
+    console.log("switch entry")
+    let fruits = 'Oranges'
+
+    switch (fruits) {
+        // case 'Oranges':
+        //     console.log("switch case 1")
+        //     break
+        case 'Mangoes':
+            console.log("switch case 2")
+            break
+        case 'Papayas':
+            console.log("switch case 3")
+            break
+    }
+
+    console.log("switch done")
+}
```

Comment:
Why these test cases are needed?

Reply:
Commented cases were recognized as bad nodes and were breaking the parse. The `getSwitchCasesAndDefault` function removes these nodes. These test cases are to represent this scenario.

Reply:
Make sense.